src/FilterPanel.tsx

```diff
@@ -0,0 +1,79 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+
+interface FilterPanelProps {
+  setFilterMode: React.Dispatch<React.SetStateAction<string>>,
+  filteredTodos: Todo[];
+}
+
+export const FilterPanel: React.FC<FilterPanelProps> = ({
+  setFilterMode, filteredTodos,
+}) => {
+  const [activeButton, setActiveButton] = useState<string>('All');
+
+  const hasCompletedTasks = filteredTodos.some((todo) => todo.completed);
+
+  const showActiveTodos = (filterName: string) => {
+    setFilterMode('Active');
+    setActiveButton(filterName);
+  };
+
+  const showAllTodos = (filterName: string) => {
+    setFilterMode('All');
+    setActiveButton(filterName);
+  };
+
+  const showCompletedTodos = (filterName: string) => {
+    setFilterMode('Completed');
+    setActiveButton(filterName);
+  };
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        3 items left
```

Comment:
The number of items left should be calculated based on the todos, not hardcoded to 3.